src/guide/security/generating-cryptographic-keys.md

```diff
@@ -0,0 +1,225 @@
+
+- `--compact` (`-c`): Specifies that the output private and public keys are displayed on separate lines and are not labeled.
+
+### Examples
+
+::: code-group Examples
+
+```bash [No parameters]
+
+# Input
+$ cargo run --bin kagami crypto
+
+# Possible Output (same layout, different keys)
+Public key (multihash): "ed01206B0F56F58761060056355DBA0E0FC489CFB2F974481ED64873082E6032796235"
+Private key (ed25519): "F71EA9D897C4338CBF4F1DC7B492AAD0BF6CE896B803D7CDB9CF25ECC15109826B0F56F58761060056355DBA0E0FC489CFB2F974481ED64873082E6032796235"
+
+```
+
+```bash [--algorithm]
+
+# Input
+$ cargo run --bin kagami crypto -a secp256k1
+
+# Possible output
+Public key (multihash): "e70121022A9D6E0D54022C0E2752E43ADD91ADA28259E1F2CE0C6D4E9183FB2882DE6749"
+Private key (secp256k1): "7687B1433FB6731E6DC635A376B3EB3B5FCD1E02C9775C1642E7FD5DA035EC75"
+
+```
+
+```bash [--seed]
+
+# Input
+$ cargo run --bin kagami crypto -s 1729
+
+# Exact output
+Public key (multihash): "ed0120B678073CFAE6E247A58B442661C7DA0E13BAC5031CBC6343EF566B8718D47D04"
+Private key (ed25519): "2669BB1099477B970E1D7D7C54E345A64A54213FCFBA2465CBCD6D4E5091A71DB678073CFAE6E247A58B442661C7DA0E13BAC5031CBC6343EF566B8718D47D04"
+
+```
+
+```bash [--private-key]
+
+# Input
+$ cargo run --bin kagami crypto -p 2669BB1099477B970E1D7D7C54E345A64A54213FCFBA2465CBCD6D4E5091A71DB678073CFAE6E247A58B442661C7DA0E13BAC5031CBC6343EF566B8718D47D04
+
+# Exact output
+Public key (multihash): "ed0120B678073CFAE6E247A58B442661C7DA0E13BAC5031CBC6343EF566B8718D47D04"
+Private key (ed25519): "2669BB1099477B970E1D7D7C54E345A64A54213FCFBA2465CBCD6D4E5091A71DB678073CFAE6E247A58B442661C7DA0E13BAC5031CBC6343EF566B8718D47D04"
+
+```
+
+```bash [--json]
+
+# Input
+$ cargo run --bin kagami crypto -j
+
+# Possible output
+{
+  "public_key": "ed0120B678073CFAE6E247A58B442661C7DA0E13BAC5031CBC6343EF566B8718D47D04",
+  "private_key": {
+    "digest_function": "ed25519",
+    "payload": "2669bb1099477b970e1d7d7c54e345a64a54213fcfba2465cbcd6d4e5091a71db678073cfae6e247a58b442661c7da0e13bac5031cbc6343ef566b8718d47d04"
+  }
+}
+
+```
+
+```bash [--compact]
+
+# Input
+$ cargo run --bin kagami crypto -c
+
+# Possible output
+ed0120B678073CFAE6E247A58B442661C7DA0E13BAC5031CBC6343EF566B8718D47D04
+2669BB1099477B970E1D7D7C54E345A64A54213FCFBA2465CBCD6D4E5091A71DB678073CFAE6E247A58B442661C7DA0E13BAC5031CBC6343EF566B8718D47D04
+ed25519
+
+```
+
+```bash [Combination of parameters]
+
+# Input
+$ cargo run --bin kagami crypto -a bls_normal -s 2048
+
+# Exact output
+Public key (multihash): "ea01610402A54ABCC40819F15E3553CC8D42D628EEAD7E1B10724BD2AFE523A7C0446EB1CB3F14D4500BD68C997784136FD056BA04215DFD2D3FDC7883B43AE94AC52B7D01525F5A80B41C01701502B46DBB9F0384CC7BE037DC2CBC928014E52A4C5C3B"
+Private key (bls_normal): "0000000000000000000000000000000035D9120A174E35E966DD92DE90B2446D4B060C8B72018B3917A1C97D7E93EAEC"
+
+```
+
+:::
+
+## Other Operations with Kagami
```

Comment:
Other operations with kagami are `kagami config`, `kagami genesis` etc.

This sections describes other ways of Kagami installation. TBH, the section seems redundant to me. Not because I know how to do it, but because it describes the things which might be applied to ANY other binary, and is out of scope of Kagami documentation. I think putting links to the guides over the internet might be enough. 

The easiest way to use Kagami for users is to use [`cargo install`](https://doc.rust-lang.org/cargo/commands/cargo-install.html), which does all the work for you:

```bash
# install from git
cargo install --git https://github.com/hyperledger/iroha/tree/iroha2-dev kagami

# or from local path
cargo install --path path/to/iroha/repo/tools/kagami

# and use it globally
kagami --help
```

Reply:
See a general task for it:

- https://github.com/hyperledger/iroha-2-docs/issues/421

Reply:
can we leave it like this for now and address it later via the issue you´ve created?